src/neighborsearch/bounded.jl

```diff
@@ -3,39 +3,122 @@
 # ------------------------------------------------------------------
 
 """
-    BoundedSearch(method, nmax)
+    BoundedSearch(method, maxneighbors=Inf; maxpercategory,
+    maxperoctant, ordermetric=Euclidean())
 
-A method for searching at most `nmax` neighbors using `method`.
+A method for searching at most `maxneighbors` neighbors using `method`. Extra
+restrictions available: `maxpercategory` and `maxperoctant`. The priority is
+given to the nearest neighbor using `ordermetric`.
 """
 struct BoundedSearch{M<:NeighborSearchMethod} <: BoundedNeighborSearchMethod
   method::M
-  nmax::Int
+  maxneighbors::Int
+  maxpercategory::Dict{Symbol,Int}
+  maxperoctant::Int
+  ordermetric::Metric
+
+  function BoundedSearch{M}(method::M, maxneighbors=0; maxpercategory=Dict(),
+                            maxperoctant=0, ordermetric=Euclidean()) where {M}
```

Comment:
We don't usually add default values in inner constructors, nor specify the types of the arguments. At least that is the patterns I've seen most often out there. Could you please rewrite this as:

```julia
function BoundedSearch{M}(method, maxneighbors, maxpercategory, maxperoctant, ordermetric) where {M}
```

and then move the logic about default values and keywords to outer constructors? This is the pattern we are following the rest of the codebase.